src/game/game.cpp

```diff
@@ -77,9 +100,15 @@ Game Game::simulate(
   ranges::for_each(defenders, [&](Defender &defender) mutable {
     if (auto attacker_index =
             defender.get_nearest_attacker_index(prev_state_attackers)) {
-      if ((defender.is_in_range(attackers[*attacker_index])) &&
-          ((defender.is_aerial_type()) ||
-           (!attackers[*attacker_index].is_aerial_type()))) {
+
+      bool should_attack = (defender.is_in_range(attackers[*attacker_index])) &&
+                           ((defender.is_aerial_type()) ||
+                            (!attackers[*attacker_index].is_aerial_type()));
+
+      should_attack =
+          should_attack && !attackers[*attacker_index].is_ability_active();
+
+      if (should_attack) {
```

Comment:
instead of all this u can just put a check in the attack functions or take damage functions to make sure that attacker dosent take damage when its ability is activated